### PR TITLE
MWPW-140268 [MILO][MEP] Overhaul MEP Action List

### PR DESCRIPTION
**Pull request by vgoodric**
Note: Don't be scared by the number of files changed.  It's really a tiny change to fragment.js and a lot of changes to personalization.js.
There are 3 files added to the "mep" folder to be able to test useBlockCode and insertScript actions.
The rest are from reorganizing the test files to make them more manageable.

Overhaul list of actions to be more powerful and simpler to use:
Replace but provide retroactive support for these actions:
1. replaceContent
2. replaceFragment
3. removeContent
4. insertContentBefore
5. insertContentAfter

And replace them with the following actions that can accept CSS Selectors and fragment locations:
 1. replace
 2.  remove
 3. insertBefore
 4. insertAfter

Add these new actions:
 1. appendToSection
 2. prependToSection

And keep these actions:
1. replacePage
2. useBlockCode
3. insertScript
4. updateMetaData

Related features:
1. @chrischrischris is adding a simplified selector feature in a separate PR.  But because appending and prepending to blocks is problematic, appendToSection and prependToSection will only work on sections and only accept simplified selectors like "section1", "section2" etc.
2. remove and replace need extra care when used on fragments.  Multiple actions on the same fragment selector cannot be performed because the first action should remove the fragment.  Similarly, if an action is supposed to remove or replace a fragment resulting from a previous replace action, the actions must "chain".
3. MEP Manifest Template has been updated but will not be uploaded until this PR is merged.

Resolves: [MWPW-140268](https://jira.corp.adobe.com/browse/MWPW-140268)

**Test URLs:**
New and complete action list:
Before: https://main--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/?martech=off
After: https://mepmoreactions--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/?martech=off
new-actions

Previous action list support:
Before: https://main--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/?actions=old&martech=off
After: https://mepmoreactions--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/?actions=old&martech=off
Note that nothing is prepended or appended to section 1 and only replaceFragment worked on section 3:
old-actions

Chaining fragment action use cases:
Before: https://main--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/chaining/?martech=off
After: https://mepmoreactions--milo--adobecom.hlx.page/drafts/vgoodrich/fragments/2024/q1/mepmoreactions/chaining/?martech=off
Note you can use the MEP button to change which manifests are on to see the layering effect.  But I've looked at every combination of the 3 manifests being on and off, and mapped the results here: https://adobe.sharepoint.com/:x:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7BE079DF0D-23F7-4424-94BD-314669060AF3%7D&file=Book.xlsx&action=default&mobileredirect=true
Stuff to notice when looking at the spreadsheet:
1. 1a and 2a behave the same.  1b and 2b behave the same. We were trying to achieve parity between a doc starting with a fragment or content.
2. since only scenario 4 is touched by the test manifest, scenario 4 has a unique outcome for every combination.  Each combination gets the exact outcome you would expect.

### CODE REVIEW

File: libs/features/personalization/personalization.js
Comment (3ch023) [inline libs/features/personalization/personalization.js:164]: CSS selector would not have slashes, so, imho, checking for '/fragments/' should be enough.
Suggestion:
```
const selectorType = (selector) => selector?.includes('/fragments/') ? 'fragment' : 'css';
```

i

**Review by 3ch023 — APPROVED** [review]
looks good, left a few minor comments

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js:159]: You're right, I was probably being overly cautious. Updated to ternary.
```
function checkSelectorType(selector) {
  return selector?.includes('/fragments/') ? 'fragment' : 'css';
}
```

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: agreed. Updated.

File: libs/mep/mepmoreactions/quote/quote.css
Comment (vgoodric) [reply libs/mep/mepmoreactions/quote/quote.css:1]: It's not for a unit test, it's for being able to provide a before and after with the useBlockCode action.  And that is where we decided "test code" will go.  I'll be able to do a fast follow (as we would once an A/B test is over) to remove this.  However, I wanted to demonstrate my new code didn't break any existing actions.

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js:267]: This if statement is trying to cover "appendtosection" and "prependtosection".  So I'm testing against the string they share (but is not present in any other actions).

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: Added a small update to default to section1 if they do not specify.
And agreed, we'll need to merge the logic.  I had a bit of a chicken and egg situation.  I knew your code would handle this better, but I needed something in place until I had your code since we decided to not allow CSS selectors for this action.
How ready is your PR?  I think we can merge this PR first and then I help if needed to merge my code into yours.  Or you submit your PR to merge into this branch and we get it working together before we merge this one.  I'm fine with either approach.

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js:660]: Correct, `mep` is just `config.mep`. I don't think I can reference `config.mep?.preview` from `config.mep.handleFragmentCommand` (but possible I'm wrong).  So the choice was calling getConfig when this function is only called from fragments.js right after it had just called that function and extrapolated mep.  But if you would prefer I call getConfig or have a better solution, I'm happy to update.

File: libs/features/personalization/personalization.js
Comment (chrischrischris) [reply libs/features/personalization/personalization.js:267]: I think in this case it'd be clearer to be more verbose and use
```suggestion
    if ((action.includes('appendtosection') || action.includes('prependtosection') ) && selector.includes('section')) {
```

File: libs/features/personalization/personalization.js
Comment (vgoodric) [reply libs/features/personalization/personalization.js]: updated based on our convo in Slack

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**
